### PR TITLE
Apply `cmake-format` on `.cmake.in` files in CI

### LINKED ISSUE #7594: Adapt CI stage to apply cmake-format on `.cmake.in` files
We currently do not format the `cmake.in` files as the action we use to check this uses hard coded commands to find the files in the project. 

https://github.com/puneetmatharu/cmake-format-lint-action/pull/9 tries to propose changes to the action we use

### PR DESCRIPTION
**Pull request by JBludau**
Fixes #7594
~~Depends on https://github.com/puneetmatharu/cmake-format-lint-action/pull/9/files~~
Depends on v1.0.5 of https://github.com/puneetmatharu/cmake-format-lint-action

- [x] This might lead to the action bumping version which needs to be included here

### CODE REVIEW

**Comment by JBludau** [comment]
the cmake-format action now also reports formatting `.cmake.in` files https://github.com/kokkos/kokkos/actions/runs/12359086031/job/34491184020?pr=7602

File: .github/workflows/cmake-format-check.yml [hunk 1]
```diff
@@ -11,8 +11,9 @@ jobs:
     steps:
     - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
     - name: cmake-format lint action
-      uses: puneetmatharu/cmake-format-lint-action@efbb497b2a8badd2c9dc638faaf8ef4a9aa71bc8 # v1.0.4
+      uses: puneetmatharu/cmake-format-lint-action@2ac346e79e7ceac958bc637c1391285fb335ed7c # v1.0.5
       with:
         args: --config-files .cmake-format.py --in-place
+        file-regex: '(.*\.cmake$|.*\.cmake\.in$|CMakeLists.txt$)'
```
Comment (dalg24) [inline .github/workflows/cmake-format-check.yml:17]:
```suggestion
        file-regex: '(\.cmake$|\.cmake\.in$|CMakeLists.txt$)'
```

Comment (JBludau) [reply .github/workflows/cmake-format-check.yml:17]: I feel like I will never understand regex. But I think we will have to match the whole filename not just the extension that was why I was using `.*`. I tested it here https://regex101.com/ but I might still be wrong :-D

Comment (dalg24) [reply .github/workflows/cmake-format-check.yml:17]: Then your 3rd alternative is wrong, the CMakeLists.txt

Comment (dalg24) [reply .github/workflows/cmake-format-check.yml:17]: Oh I take it back.  I expected whatever gets matched will have a relative path from the root of the repo

**Review by dalg24 — APPROVED** [review]

**Review by janciesko — APPROVED** [review]

**Pull request merged**